crates/subspace-farmer/src/farming.rs

```diff
@@ -14,6 +14,8 @@ use subspace_core_primitives::{LocalChallenge, Salt};
 use subspace_rpc_primitives::{SlotInfo, Solution, SolutionResponse};
 use thiserror::Error;
 use tokio::task::JoinHandle;
+#[cfg(test)]
+use tokio::time::{sleep, Duration};
```

Comment:
Sleeps everywhere is a bad idea. There is nothing in this file that should be test-specific, otherwise you're testing imaginary scenarios that will not happen in practice.

Reply:
I think there is one:
Waiting for recommitments to finish before each call to `find_by_range` would not be ideal. Because in this case, if commitment creation will take some time, the farmer will not be able to solve the slot. 
That part is specifically designed for test purpose, and that part requires tokio sleep and duration. The reason for sleep is race between channel message retrieval, so that it won't hang forever.

Reply:
Recommitment finishing is a clear deterministic event, you should neither modify this file nor have timeouts to know that it happened. Having something with `#[cfg(test)]` is an acceptable idea in some narrow cases, but as we patch multiple files just to make tests pass then either architecture is bad or tests are bad or both, but patching files with special-cases for tests is not a sustainable long-term solution either way.